biz.aQute.repository/src/aQute/bnd/repository/osgi/OSGiRepository.java

```diff
@@ -331,4 +332,12 @@ public void close() throws IOException {
 			throw Exceptions.duck(e);
 		}
 	}
+
+	public static String getRepositoryName(Repository repo) {
```

Comment:
This seems an invalid place to put this method. It has nothing to do with the OSGiRepository type. This is bad coupling.

And it is only called from one place. I would put the logic there.